Enumerator panics on windows when removing a device mid enumeration

panic: runtime error: index out of range [0] with length 0
```
goroutine 883 [running]:\ngo.bug.st/serial/enumerator.retrievePortDetailsFromDevInfo(0xc00011f3b0, 0xc000121560, 0x5, 0x0)
/go/pkg/mod/go.bug.st/serial@v1.3.5/enumerator/usb_windows.go:351 +0x365
go.bug.st/serial/enumerator.nativeGetDetailedPortsList(0x0, 0x0, 0x0, 0x0, 0x0)
/go/pkg/mod/go.bug.st/serial@v1.3.5/enumerator/usb_windows.go:299 +0x1b1
go.bug.st/serial/enumerator.GetDetailedPortsList(...)
/go/pkg/mod/go.bug.st/serial@v1.3.5/enumerator/enumerator.go:31
```
I suspect N needs a zero check on line 350:
![image](https://user-images.githubusercontent.com/11500738/167532184-9918c7a1-495c-4d26-9bdc-fb42034b63ed.png)

--- a/enumerator/usb_windows.go
+++ b/enumerator/usb_windows.go
@@ -347,10 +347,12 @@ func retrievePortDetailsFromDevInfo(device *deviceInfo, details *PortDetails) er
 		the result of spdrpFriendlyName is therefore unique and suitable as an alternative string to for a port choice */
 	n := uint32(0)
 	setupDiGetDeviceRegistryProperty(device.set, &device.data, spdrpFriendlyName /* spdrpDeviceDesc */, nil, nil, 0, &n)
-	buff := make([]uint16, n*2)
-	buffP := (*byte)(unsafe.Pointer(&buff[0]))
-	if setupDiGetDeviceRegistryProperty(device.set, &device.data, spdrpFriendlyName /* spdrpDeviceDesc */, nil, buffP, n, &n) {
-		details.Product = syscall.UTF16ToString(buff[:])
+	if n > 0 {
+		buff := make([]uint16, n*2)
+		buffP := (*byte)(unsafe.Pointer(&buff[0]))
+		if setupDiGetDeviceRegistryProperty(device.set, &device.data, spdrpFriendlyName /* spdrpDeviceDesc */, nil, buffP, n, &n) {
+			details.Product = syscall.UTF16ToString(buff[:])
+		}
 	}
 
 	return nil

